x-pack/plugins/security_solution/public/detection_engine/rule_response_actions/endpoint/overwrite_process_field.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { UseField } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { ToggleField } from '@kbn/es-ui-shared-plugin/static/forms/components';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface OverwriteFieldProps {
+  path: string;
+  disabled: boolean;
+  readDefaultValueOnForm: boolean;
+}
+
+const CONFIG = {
```

Comment:
I don't think using a `singleton` across all instances of `OverwriteFieldComponent` is a good approach. I suggest you move it to be a memoized value inside of the component instead. Doing so will also likely remove the need for you to do the type casting and should be able to remove the `eslint` rule disablement